### PR TITLE
Link Enhancement - Document & Video actions

### PR DESCRIPTION
**Pull request by hcarneiro**
- [x] `Fliplet.Navigate` needs to be enhanced to allow `document` and `video` actions
- [x] Changes (perhaps to `fliplet-studio`) needs to be made to allow multiple levels of `inline` and `fixed` providers

### CODE REVIEW

File: interface.html [hunk 1]
```diff
@@ -121,6 +123,56 @@
     </div>
   </div>
 
+  <div id="documentSection" class="section">
```
Comment (squallstar) [inline interface.html:126]: indentation here looks wrong

File: js/interface.js [hunk 2]
```diff
@@ -6,12 +6,95 @@ var fields = [
...
+var providerInstance;
+
+var currentMode = null;
+
+var files = widgetInstanceData.files || {
+  'document': {
+      selectFiles: {},
+      selectMultiple: false,
+      type: 'document',
+      fileExtension: ['PDF', 'DOC', 'DOCX', 'KEY', 'PPT', 'ODT', 'XLS', 'XLSX']
```
Comment (squallstar) [inline js/interface.js]: can we have all of them as lowercase please?

Comment (squallstar) [reply js/interface.js]: also, why do you need both the type and fileExtension?

File: interface.html (hunk 1 above)
Comment (hcarneiro) [reply interface.html:126]: Fixed

File: js/interface.js (hunk 2 above)
Comment (hcarneiro) [reply js/interface.js]: Yeah, I don't need it. Removed!

**Pull request merged**